pylearn2/termination_criteria/__init__.py

```diff
@@ -221,16 +221,16 @@ def initialize(self, model):
         if self._new_epochs:
             self._epochs_done = 0
```

Comment:
If `push_monitor` is called with `transfer_experience=True`, then it will work.
Regarding the separate counter, I would agree, counting the number of times `EpochCounter.continue_learning()` has been called is not great, especially if we call it once before the training even starts.